creator-node/src/utils.js

```diff
@@ -484,20 +486,22 @@ async function createDirForFile (fileStoragePath) {
   await fs.ensureDir(dir)
 }
 
-async function writeStreamToFileSystem (stream, expectedStoragePath, createDir = false) {
+async function writeStreamToFileSystem (inputStream, expectedStoragePath, createDir = false) {
   if (createDir) {
     await createDirForFile(expectedStoragePath)
   }
 
-  const destinationStream = fs.createWriteStream(expectedStoragePath)
-  stream.pipe(destinationStream)
-  return new Promise((resolve, reject) => {
-    destinationStream.on('finish', () => {
-      resolve()
-    })
-    destinationStream.on('error', err => { reject(err) })
-    stream.on('error', err => { destinationStream.end(); reject(err) })
-  })
+  await streamFileToDiskHelper(inputStream, expectedStoragePath)
+}
+
+async function streamFileToDiskHelper (inputStream, expectedStoragePath) {
+  // https://nodejs.org/en/docs/guides/backpressuring-in-streams/
+  // Cleaner way to handle piping data between streams since this handles all
+  // events such as finish, error, end etc in addition to being async/awaited
+  await pipeline(
```

Comment:
do we need to return here or on line 494 to be consistent with previous pattern

Reply:
shouldnt matter since return val is not consumed anywhere, just flagging

Reply:
we don't need the return since an async function implicitly returns a promise

Reply:
can we add an err handling callback? in the docs example:
```
  (err) => {
    if (err) {
      console.error('Pipeline failed.', err);
    } else {
      console.log('Pipeline succeeded.');
    }
  }
```

Reply:
i intentionally wanted this error to be propagated up @vicky-g

Reply:
> we don't need the return since an async function implicitly returns a promise

ya i meant the return value of `await pipeline`, i know promise is returned but the return value is not propagated
anyway doesnt matter since not consumed anywhere